lib/roadie/rails/asset_pipeline_provider.rb

```diff
@@ -12,11 +14,23 @@ def initialize(pipeline)
 
       def find_stylesheet(name)
         if (asset = find_asset_in_pipeline(name))
-          Stylesheet.new("#{asset.pathname} (live compiled)", asset.to_s)
+          Stylesheet.new("#{filename(asset)} (live compiled)", asset.to_s)
         end
       end
 
       private
+      def sprockets_4_or_later?
+        Sprockets::VERSION.split('.')[0].to_i > 3
+      end
+
+      def filename(asset)
+        if sprockets_4_or_later?
```

Comment:
I think it makes more sense to use a duck and check `respond_to?(:filename)` instead of an explicit version check. Would you agree?